src/main/java/de/jtheb/fh/lagerverwaltung/entities/Compartment.java

```diff
@@ -42,19 +42,26 @@ public int getItemsVolume() {
         return volume;
     }
 
-    public boolean itemFits(Item item) {
-        this.getArticleNr();
+    public boolean itemFitsWithArticleNr(Item item) {
+        return (this.articleNrFits(item) && itemFitsVolume(item));
+    }
+
+    public boolean itemFitsVolume(Item item) {
         return (this.getVolume() - this.getItemsVolume() - item.getVolume() >= 0);
     }
 
     public boolean isFull() {
```

Comment:
I believe that the method `isFull` is not good to have, except it's needed for some UI stuff. It's unlikely that a shelve will be completely full, so that the remaining space is 0. We will be using the itemFits method most of the time.